pos_partner_sale_warning/static/src/js/ProductScreen.esm.js

```diff
@@ -0,0 +1,28 @@
+/** @odoo-module **/
+
+import PartnerListScreen from "point_of_sale.PartnerListScreen";
+import Registries from "point_of_sale.Registries";
+
+export const PartnerListScreenWarning = (PartnerListScreen) =>
+    class extends PartnerListScreen {
+        clickPartner(partner) {
+            if (
+                this.state.selectedPartner &&
+                this.state.selectedPartner.id === partner.id
+            ) {
+                return super.clickPartner(partner);
+            }
+            if (partner && partner.sale_warn && partner.sale_warn !== "no-message") {
+                this.showPopup("ErrorPopup", {
+                    title: this.env._t(`Warning for ${partner.name}`),
+                    body: this.env._t(partner.sale_warn_msg),
+                });
+                if (partner.sale_warn === "block") {
+                    return;
+                }
```

Comment:
What does it change? Why don't return super?


```suggestion

```


https://github.com/odoo/odoo/blob/c1830f569b614530c5b61859b99f4e0c7caad88b/addons/point_of_sale/static/src/js/Screens/PartnerListScreen/PartnerListScreen.js#L143-L150